server/raft.go

```diff
@@ -4228,7 +4228,9 @@ func (n *raft) switchToCandidate() {
 	defer n.Unlock()
 
 	// If we are catching up or are in observer mode we can not switch.
-	if n.observer || n.paused {
+	// Avoid petitioning to become leader if we're behind on applies.
+	if n.observer || n.paused || n.applied < n.commit {
+		n.resetElect(minElectionTimeout / 4)
```

Comment:
Why the div 4?

Reply:
Don't want to slow down the potential for an election too much, so this ends up being 1 second in normal operation instead of 4 to 9 seconds. Divide because then it speeds up in the unit tests along with all the other consts.

Reply:
ok